pkg/envoy/ads/server.go

```diff
@@ -70,23 +73,28 @@ func (s *Server) withXdsLogMutex(f func()) {
 
 // Start starts the ADS server
 func (s *Server) Start(ctx context.Context, cancel context.CancelFunc, port int, adsCert *certificate.Certificate) error {
-	grpcServer, lis, err := utils.NewGrpc(ServerType, port, adsCert.GetCertificateChain(), adsCert.GetPrivateKey(), adsCert.GetTrustedCAs())
+	grpcServer, lis, err := utils.NewGrpc(ServerType, port, xdsServerCertificateCommonName, s.certManager)
 	if err != nil {
 		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrStartingADSServer)).
-			Msg("Error starting ADS server")
+			Msg("Error creating ADS server")
```

Comment:
The error code refers to an error starting the ADS server, while the log refers to an error creating the server. Let's make this consistent.

Further, since `utils.NewGrpc` already logs an error, let's skip logging an error here and return by wrapping the original error:
`return fmt.Errorf("error starting ADS server: %w, err)`